### PR TITLE
Fixing COLORS.GRAY

### LINKED ISSUE #12536: No mapping to "gray" in COLORS constants
**Describe the bug**
Missing(possibly deprecated) color value in COLORS constants

**Steps to reproduce (REQUIRED)**
Steps to reproduce the behavior, libraries used with version number, and/or any setup information to easily reproduce:

1. Go to `https://github.com/MetaMask/metamask-extension/blob/develop/ui/helpers/constants/design-system.js#L13`
2. Go to `https://github.com/MetaMask/metamask-extension/blob/develop/ui/css/design-system/colors.scss#L113-L142`
3. See missing `GRAY: gray` value

**Expected behavior**
Either add or remove `GRAY` value from `design-system.js`

**Screenshots**
Screen Shot 2021-10-29 at 9 23 03 AM

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/MetaMask/metamask-extension/issues/12536

### CODE REVIEW

File: ui/css/design-system/colors.scss [hunk 1]
```diff
@@ -97,6 +97,7 @@ $success-2: #caf4d1;
 $success-3: #219e37;
 
 $ui-black: #24292e;
+$ui-grey: #d6d9dc;
```
Comment (ryanml) [inline ui/css/design-system/colors.scss:100]: Went with the value for `$Grey-100` here, for consistency with `$ui-black` (which equals `$Black-100`)

File: ui/helpers/constants/design-system.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ export const COLORS = {
   UI3: 'ui-3',
   UI4: 'ui-4',
   BLACK: 'black',
-  GRAY: 'gray',
+  GREY: 'grey',
```
Comment (ryanml) [inline ui/helpers/constants/design-system.js:13]: Spelling consistency why not

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 3]
```diff
@@ -495,7 +495,7 @@ export default class ConfirmTransactionBase extends Component {
                     <Typography
                       variant={TYPOGRAPHY.H7}
                       fontStyle={FONT_STYLE.ITALIC}
-                      color={COLORS.GRAY}
+                      color={COLORS.UI4}
```
Comment (ryanml) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:498]: This was the only use of `COLORS.GRAY` in the codebase, it fell back to and inherited `COLORS.UI4` already so nothing changed here

Comment (georgewrmarshall) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:498]: nice

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Review by mcmire — APPROVED** [review]
We should consider figuring out a way to generate the colors in Sass land from the colors in JS land. I know it's relatively easy to do that if you're using Webpack, but I'm not so sure about Browserify. Regardless, this makes sense to me 👍🏻

**Pull request merged**